sdk/v2/core/events.go

```diff
@@ -0,0 +1,361 @@
+package core
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/brigadecore/brigade/sdk/v2/internal/restmachinery"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+)
+
+// EventsSelector represents useful filter criteria when selecting multiple
+// Events for API group operations like list, cancel, or delete.
+type EventsSelector struct {
```

Comment:
In my head, things were ordered according to a list retrieval workflow... so you have selector types and options types (request), followed by lists and core objects (response), and then finally the interface (last because the bulk of the code is the interface's implementation). That was admittedly pretty arbitrary, but was at least consistently applied.

But I can clearly see the rationale in ordering them by relative importance and I think that probably makes more sense, so I'm happy to rearrange.

Reply:
Compromised on this... moved structs like `Event` before things like `EventList` and `EventsSelector`, but left interfaces like `EventsClient` and structs like `eventsClient` farther down, because the bulk of each file is implementation of those.